pkgmgmtutils/utils.go

```diff
@@ -22,6 +25,73 @@ type sha256Outputs struct {
 	windowsSha []byte // sha256 for <ctl>-windows binary
 }
 
+// getGitHubSha256 extracts the sha256 strings from existing .sha256 files created as part of the build process.
+// Those .sha256 files need to be located in the GitHub Release for this version.
+// It returns the sha256s and any read errors encountered. It will also return ErrNoSha256sFound if any of the platform
+// shas are found.
+func getGitHubSha256(assets []github.ReleaseAsset, reShaFilename string) (*sha256Outputs, error) {
+	if reShaFilename == "" {
+		return nil, nil // special case to indicate that cli sha256s are not needed
+	}
+
+	// Scan outputDir directory looking for any files that match the reOS regular expression as targets for extraction
+	reOS := regexp.MustCompile(reShaFilename)
+
+	shas := sha256Outputs{}
+	for _, f := range assets {
+		s := reOS.FindStringSubmatch(f.GetName())
+		if s == nil {
+			continue
+		}
+
+		var err error
+
+		switch s[1] {
+		case "darwin":
+			shas.darwinSha, err = extractShaFromURL(f.GetBrowserDownloadURL())
+		case "linux":
+			shas.linuxSha, err = extractShaFromURL(f.GetBrowserDownloadURL())
+		case "windows":
+			shas.windowsSha, err = extractShaFromURL(f.GetBrowserDownloadURL())
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+	if shas.darwinSha == nil && shas.linuxSha == nil && shas.windowsSha == nil {
```

Comment:
I think this should be `||` since we want to error if any of them aren't found

Reply:
I wasn't sure if we'd ever have cases where we'd only have 1 or 2 OSes provided, hence why the logic is testing for at least one. I can change logic and error message to reflect all three *must* be present if that's what we want.